### PR TITLE
add OS X support to open terminal here

### PR DESCRIPTION
**Pull request by spdegabrielle**

### CODE REVIEW

File: scripts/open-terminal.rkt [hunk 1]
```diff
@@ -3,17 +3,29 @@
...
-                           " --working-directory=\"" dir "\""
-                           " -t \"" dir "\""
-                           "&"))
-    #f))
+  #:os-types (unix macosx)
+  #:output-to message-box
+  
+  (cond
 
+    [(equal? (system-type 'os) 'unix)
```
Comment (Metaxal) [inline scripts/open-terminal.rkt]: Here it's better to use 
```racket
(case (system-type 'os)
  [(unix) ...]
  [(macosx) ...]
  [(windows) ...])
```

**Review by Metaxal — COMMENTED** [review]
Do you think you could use the opportunity to add the same thing for windows, which seems to be something like `(system (format "start ~s" dir))`. Not sure if you can test it though, but maybe coordinate with someone on slack?

If it's too complicated, just update the script help-string to say linux and macosx only.

**Comment by spdegabrielle** [comment]
I don't have a windows machine to test. :(

File: scripts/open-terminal.rkt (hunk 1 above)
Comment (spdegabrielle) [reply scripts/open-terminal.rkt]: done!

**Comment by spdegabrielle** [comment]
got no way of testing the windows solution

```
#lang racket/base
(require racket/system
         racket/path
         quickscript)

(script-help-string "Open a terminal in the current directory.")

(define-script open-terminal
  #:label "Open terminal"
  #:menu-path ("&Utils")
  #:os-types (unix macosx)
  #:output-to message-box

  (case (system-type 'os)
    [(unix) (λ (str #:file f)  
              (define dir (path->string (path-only f)))
              (system (string-append "gnome-terminal"
                                     " --working-directory=\"" dir "\""
                                     " -t \"" dir "\""
                                     "&"))
              #f)]
    [(macosx) (λ (str #:file f)  
                (define dir (path->string (path-only f)))
                (define osascriptdir (string-append
                                      (path->string (find-system-path 'pref-dir))
                                      "quickscript/user-scripts/"))
                (system (string-append "osascript -e 'tell app \"Terminal\" to do script \"cd \\\"" dir "\\\"\"'" ))
                #f)]
    ;[(windows) (system (string-append "cmd /c start cmd.exe /K \"cd " dir))]
    ))

```

**Comment by spdegabrielle** [comment]
forgot to run the script

**Comment by Metaxal** [comment]
Can you update the help string to say "(linux and macosx only)" ?

**Comment by spdegabrielle** [comment]
`make-readme.sh` done and added to pr

**Comment by spdegabrielle** [comment]
hold on - bert is helping

**Comment by spdegabrielle** [comment]
Thank you @bdeket for doing windows code

**Comment by Metaxal** [comment]
Great, thanks a lot Stephen!

File: scripts/open-terminal.rkt [hunk 2]
```diff
@@ -3,17 +3,30 @@
...
   #:menu-path ("&Utils")
-  #:os-types (unix)
-  (λ (str #:file f)  
-    (define dir (path->string (path-only f)))
-    (system (string-append "gnome-terminal"
-                           " --working-directory=\"" dir "\""
-                           " -t \"" dir "\""
-                           "&"))
-    #f))
+  #:os-types (unix macosx windows)
```
Comment (Metaxal) [inline scripts/open-terminal.rkt:11]: This is faulty: it should be `macos`. Didn't you compile and run the script before submitting? It should tell you there's an error.

When you make a PR (for any Racket project), always do the following:
- raco pkg setup [package]
- raco test [package]

File: scripts/open-terminal.rkt [hunk 3]
```diff
@@ -3,17 +3,30 @@
...
+    [(macosx)
+     (λ (str #:file f)  
+       (define dir (path->string (path-only f)))
+       (system
+        (string-append "osascript -e 'tell app \"Terminal\" to do script \"cd \\\"" dir "\\\"\"'" ))
+       #f)]
+    [(windows)
+     (λ (str #:file f)
+       (define dir (path->string (path-only f)))
+       (shell-execute #f "cmd.exe" "" dir 'sw_shownormal))]))
```
Comment (Metaxal) [inline scripts/open-terminal.rkt:32]: missing newline here. Do you have time to update soon? Otherwise I'll do it.

**Review by Metaxal — COMMENTED** [review]
Hi Stephen, turns out it doesn't compile properly.
Do you have time to update soon? Otherwise I'll do it.

File: scripts/open-terminal.rkt (hunk 2 above)
Comment (spdegabrielle) [reply scripts/open-terminal.rkt:11]: caught out! 
Sorry saw this as work today - I will fix tomorrow.

**Pull request merged**